### PR TITLE
Connect to API 🔗

### PR DESCRIPTION
**Pull request by rmrmrmr**
## On this pull request we: 
- Implemented the createAsyncThunk component to connect via Fetch to the API
- Create different files to organize the redux communications.
- Create a function to convert the object to a readable array for the list of books.
- All Linters passed.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,50 +1,30 @@
...
+      status: 'failed',
     }),
-  },
-  extraReducers: (builder) => {
-    builder.addDefaultCase((state) => state);
   },
 });
 
-export const { addBook, removeBook } = books.actions;
+// export const { addBook, removeBook } = books.actions;
```
Comment (DammyShittu) [inline src/redux/books/books.js]: - [x] Kindly remove commented code.

File: src/components/AddBook.js [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
+  const newBook = async (e) => {
     e.preventDefault();
     const { target } = e;
     const title = target.title.value;
     const author = target.author.value;
-    dispatch(addBook(createBook(title, author)));
+    dispatch(addBooksThunk(createBook(title, author)));
     target.title.value = '';
     target.author.value = '';
   };
```
Comment (DammyShittu) [inline src/components/AddBook.js]: - [x] When I add a book I have to reload the page for the newly added book to reflect on the page from the API. Kindly refactor your code and ensure that when a book is added it is not only added to the API but also on the page immediately without page refresh.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @rmrmrmr,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by victorokoroji — APPROVED** [review]
Hi @rmrmrmr, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**